### PR TITLE
[WF][84942320] Alternate image servers

### PR DESCRIPTION
**Pull request by wflann**
[Alternate Image Servers in Photo Plus](https://www.pivotaltracker.com/story/show/84942320)

### CODE REVIEW

File: lib/photoplus.coffee [hunk 1]
```diff
@@ -17,7 +17,7 @@ define [
       resultSelector       : '#photo_plus_result_'
       pinSelector          : '#photo_plus_pin_'
       offset               : 5
-      imagesToLoad         : 3
+      imagesToLoad         : 4
```
Comment (wflann) [inline lib/photoplus.coffee]: I don't know if there are implications to switching this, but I wanted to even out the requests to each server.

File: lib/photoplus.coffee [hunk 2]
```diff
@@ -120,6 +122,13 @@ define [
         @select('gallerySelector').animate options, 400, =>
           @processing = false
 
+    @pickServer = (num) ->
+      val = num % 2
+      if val == 0
+        server = 'http://image.apartmentguide.com'
+      else
+        server = 'http://image1.apartmentguide.com'
```
Comment (sbrauer) [inline lib/photoplus.coffee]: What do you think about adding a `servers` array to `@defaultAttrs` with the value 
```coffee
[
    'http://image.apartmentguide.com'
    'http://image1.apartmentguide.com'
]
```

and then modifying pickServer like so?

```coffee
@pickServer = (num) ->
  @attr.servers[num % @attr.servers.length]
```
This seems cleaner to me and would make it easier to add more image servers later.

Comment (wflann) [inline lib/photoplus.coffee]: You're right. Sounds good--I'll rework.

**Comment by sbrauer** [comment]
Sorry to switch things up on you, but I looked at the story again and saw how it suggests using image-helper.js.
Thinking that might be the way to go since image-helper.js does the sort of round-robin logic we chatted about earlier (https://github.com/rentpath/image-helper.js/blob/dev/src/image-helper.coffee#L41-L44). It also pulls the image servers from an existing meta tag, so there'd be no need for this repo to hard code the server array.  
So basically you'd need to:
- update bower.json to add the dependency on image-helper.js
- require "image-helper" at the top of this file
- change the part of this file that generates the img tag to use image-helper's `url` method; something like:
```coffee
        html += "<img src='#{imageHelper.url(@addSize(photo.path))}' "
``` 
- revert other changes

Lemme know if you have questions or concerns.

File: lib/photoplus.coffee (hunk 1 above)
Comment (pasha) [inline lib/photoplus.coffee]: that number means how many photos to prefetch ahead, not sure if we need to change it.

**Comment by pasha** [comment]
:+1:  what @sbrauer said

**Comment by wflann** [comment]
Thanks guys--will redo

**Comment by sbrauer** [comment]
:+1:
